bids_prov/spm_parser.py

```diff
@@ -1,74 +1,102 @@
-import sys
-import click
+import argparse
 import json
 import os
 import re
-from difflib import SequenceMatcher
+import hashlib
 
 from collections import defaultdict
+from bids_prov import spm_load_config as conf
+from bids_prov import get_id
 
-from . import spm_load_config as conf
-from . import get_id
+def format_activity_name(activity_name: str, l_max=30) -> str:
+    """Function to limit name of activity
 
+    Parameters
+    ----------
+    activity_name : name of activity
+    l_max : maximal length with cut in words between dots
 
-def format_activity_name(s, l=30):
-    if s.startswith("spm."):
-        s = s[4:]
-    tmp = s.split(".")
-    while sum(map(len, tmp)) > l:
-        tmp = tmp[1:]
-    return ".".join(tmp)
 
+    Examples
+    --------
+    Examples should be written in doctest format, and should illustrate how
+    to use the function.
 
-def get_input_entity(left, right):
-    """get input Entity if possible else return None
+    >>> print(format_activity_name("cfg_basicio.file_dir.file_ops.file_move._1"))
+    file_dir.file_ops.file_move._1
 
-    left: string
-        left side of ' = '
-    right: string
-        right side of ' = '
     """
-    if conf.has_parameter(left):
-        return None
-    if not next(re.finditer(conf.PATH_REGEX, right), None):
-        return None
+    # s example : cfg_basicio.file_dir.file_ops.file_move._1
+    if activity_name.startswith("spm."):
+        activity_name = activity_name[4:]
+    act_split = activity_name.split(".")  # ['cfg_basicio', 'file_dir', 'file_ops', 'file_move', '_1']
+    while sum(map(len, act_split)) > l_max:  # sum of the lengths of each element of tmp
+        act_split = act_split[1:]
+    return ".".join(act_split)  # file_dir.file_ops.file_move._1
+
+
+def get_input_entity(right: str, verbose=False) ->  list:
+    """Get input Entity if possible else return None
```

Comment:
Can you explicit/give an example of when it will be possible versus not?

Reply:
Done